docs/src/userguide/language_basics.rst

```diff
@@ -581,44 +906,93 @@ be careful what you do.
 Type Casting
 ------------
 
-Where C uses ``"("`` and ``")"``, Cython uses ``"<"`` and ``">"``. For example::
+Cython language supports type casting in a simmilar way as C. Where C uses ``"("`` and ``")"``,
+Cython uses ``"<"`` and ``">"``. In pure python mode, ``cast()`` function is used. For example:
+
+.. tabs::
+
+    .. group-tab:: Pure Python
+
+        .. code-block:: python
+
+            def main():
+                p: cython.p_char
+                q: cython.p_float
+                p = cast(cython.p_char, q)
+
+        When casting a C value to a Python object type or vice versa,
+        Cython will attempt a coercion. Simple examples are casts like ``cast(int, pyobj)``,
+        which converts a Python number to a plain C ``int`` value, or ``cast(bytes, charptr)``,
+        which copies a C ``char*`` string into a new Python bytes object.
+
+         .. note:: Cython will not prevent a redundant cast, but emits a warning for it.
 
-    cdef char *p
-    cdef float *q
-    p = <char*>q
+        To get the address of some Python object, use a cast to a pointer type
+        like ``cast(p_void, ...)`` or ``cast(pointer(PyObject), ...)``.
+        You can also cast a C pointer back to a Python object reference
+        with ``cast(object, ...)``, or a more specific builtin or extension type
+        (e.g. ``cast(MyExtType, ptr)``). This will increase the reference count of
+        the object by one, i.e. the cast returns an owned reference.
+        Here is an example:
```

Comment:
The only place as far where you should not move test, yeah it'll be better it you could turn off tabs but will left finctionality of changing content. (I think it's good issue for sphinx-tabs repo)